lib/src/replay/element_parsers/element_object_parser.dart

```diff
@@ -1,12 +1,49 @@
 import 'package:flutter/cupertino.dart';
+import 'package:flutter/rendering.dart';
 import 'package:posthog_flutter/src/replay/element_parsers/element_data.dart';
+import 'package:posthog_flutter/src/replay/element_parsers/element_parser.dart';
 import 'package:posthog_flutter/src/replay/mask/posthog_mask_controller.dart';
+import 'package:posthog_flutter/src/replay/mask/posthog_nomask_widget.dart';
 
 class ElementObjectParser {
   ElementData? relateRenderObject(
     ElementData activeElementData,
     Element element,
   ) {
+    if (element.widget is PostHogNoMaskWidget) {
+      final elementData = ElementParser().relate(element, activeElementData);
+
+      if (elementData != null) {
+        activeElementData.addChildren(elementData);
+        return elementData;
+      }
+    }
+
+    if (element.widget is Text) {
+      final elementData = ElementParser().relate(element, activeElementData);
+
+      if (elementData != null) {
+        activeElementData.addChildren(elementData);
+        return elementData;
+      }
+    }
+
+    if (element.renderObject is RenderImage) {
+      final String dataType = element.renderObject.runtimeType.toString();
+
+      final parser = PostHogMaskController.instance.parsers[dataType];
+      if (parser != null) {
+        final elementData = parser.relate(element, activeElementData);
+
+        if (elementData != null) {
+          activeElementData.addChildren(elementData);
+          return elementData;
+        }
+      }
+    }
```

Comment:
i think this can be a `when/switch` case since each item returns anyway, or is it possible to go thru multiple `if`s for the same component?

Reply:
I think not because with `when/switch` in dart doesn't have a native switch for type checks